src/main/java/com/denizenscript/ddiscordbot/events/DiscordCommandAutocompleteScriptEvent.java

```diff
@@ -0,0 +1,109 @@
+import net.dv8tion.jda.api.interactions.commands.CommandInteractionPayload;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DiscordCommandAutocompleteScriptEvent extends DiscordCommandInteractionScriptEvent {
+
+    // <--[event]
+    // @Events
+    // discord command autocomplete
+    //
+    // @Switch for:<bot> to only process the event for a specified Discord bot.
+    // @Switch channel:<channel_id> to only process the event when it occurs in a specified Discord channel.
+    // @Switch group:<group_id> to only process the event for a specified Discord group.
+    // @Switch name:<command_name> to only process the event for a specified Discord application command. Spaces are replaced with underscores.
+    // @Switch option:<option_name> to only process the event for a specified autocompletable option.
+    //
+    // @Triggers when a Discord user queries a slash command option that can be autocompleted.
+    //
+    // @Plugin dDiscordBot
+    //
+    // @Group Discord
+    //
+    // @Context
+    // <context.bot> returns the relevant DiscordBotTag.
+    // <context.channel> returns the DiscordChannelTag.
+    // <context.group> returns the DiscordGroupTag.
+    // <context.interaction> returns the DiscordInteractionTag.
+    // <context.command> returns the DiscordCommandTag.
+    // <context.options> returns the supplied options as a MapTag.
+    // <context.focused_option> returns the name of the focused option.
+    //
+    // @Determine
+    // ListTag to suggest values to the Discord client. Each entry can be an ElementTag which controls both the value and display of the choice or a MapTag with "name" and "value" keys to control both separately.
+    //
+    // -->
+
+    public static DiscordCommandAutocompleteScriptEvent instance;
+
+    public DiscordCommandAutocompleteScriptEvent() {
+        instance = this;
+        registerCouldMatcher("discord command autocomplete");
+        registerSwitches("option");
+    }
+
+    public CommandAutoCompleteInteractionEvent getAutocompleteEvent() {
+        return (CommandAutoCompleteInteractionEvent) event;
+    }
+
+    @Override
+    public CommandInteractionPayload getPayload() {
+        return getAutocompleteEvent().getInteraction();
+    }
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (!runGenericSwitchCheck(path, "option", getAutocompleteEvent().getFocusedOption().getName())) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        switch (name) {
+            case "focused_option":
+                return new ElementTag(getAutocompleteEvent().getFocusedOption().getName(), true);
+        }
+        return super.getContext(name);
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determination) {
+        if (!determination.canBeType(ListTag.class)) {
```

Comment:
Not a stable check - *anything* can be a ListTag.
The determination should probably just be prefixed (it's intended in the future for determinations with prefixes to have native handling, so those are preferable anyway)